src/query/helper.rs

```diff
@@ -149,6 +152,61 @@ pub trait QuerySelect: Sized {
         self
     }
 
+    /// Add a DISTINCT expression
+    /// ```
+    /// use sea_orm::{entity::*, query::*, tests_cfg::cake, DbBackend};
+    /// struct Input {
+    ///     name: Option<String>,
+    /// }
+    /// let input = Input {
+    ///     name: Some("cheese".to_owned()),
+    /// };
+    /// assert_eq!(
+    ///     cake::Entity::find()
+    ///         .filter(
+    ///             Condition::all().add_option(input.name.map(|n| cake::Column::Name.contains(&n)))
+    ///         )
+    ///         .distinct()
+    ///         .build(DbBackend::MySql)
+    ///         .to_string(),
+    ///     "SELECT DISTINCT `cake`.`id`, `cake`.`name` FROM `cake` WHERE `cake`.`name` LIKE '%cheese%'"
+    /// );
+    /// ```
+    fn distinct(mut self) -> Self {
+        self.query().distinct();
+        self
+    }
+
+    /// Add a DISTINCT ON expression
+    /// NOTE: this function is only supported by `sqlx-postgres`
+    /// ```
+    /// use sea_orm::{entity::*, query::*, tests_cfg::cake, DbBackend};
+    /// struct Input {
+    ///     name: Option<String>,
+    /// }
+    /// let input = Input {
+    ///     name: Some("cheese".to_owned()),
+    /// };
+    /// assert_eq!(
+    ///     cake::Entity::find()
+    ///         .filter(
+    ///             Condition::all().add_option(input.name.map(|n| cake::Column::Name.contains(&n)))
+    ///         )
+    ///         .distinct_on(cake::Column::Name)
+    ///         .build(DbBackend::Postgres)
+    ///         .to_string(),
+    ///     "SELECT DISTINCT ON (\"name\") \"cake\".\"id\", \"cake\".\"name\" FROM \"cake\" WHERE \"cake\".\"name\" LIKE '%cheese%'"
+    /// );
+    /// ```
+    #[cfg(feature = "sqlx-postgres")]
```

Comment:
We try to avoid feature gated method

Reply:
I fixed in https://github.com/SeaQL/sea-orm/pull/902/commits/c9871cb2d25de92336501799faa7eb29595ef95b